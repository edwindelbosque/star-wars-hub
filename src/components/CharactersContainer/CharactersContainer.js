import React from 'react';
import './CharactersContainer.scss';
import CharacterCard from '../CharacterCard/CharacterCard'
import PropTypes from 'prop-types';

const CharactersContainer = ({ characters }) => {
  return (
    <section className='characters-cont'>
      {characters.map((character, index) => {
        const { name, species, homeworld, relatedFilms } = character;
        return <CharacterCard
          key={index}
          name={name}
          species={species}
          homeworldName={homeworld.name}
          homeworldPop={homeworld.population}
          relatedFilms={relatedFilms}
        />
      })}
    </section>
  )
}

CharactersContainer.propTypes = {
  character: PropTypes.object,
  index: PropTypes.number,
  name: PropTypes.string,
  species: PropTypes.string,
  homeworld: PropTypes.object,
  relatedFilms: PropTypes.array,
};

export default CharactersContainer