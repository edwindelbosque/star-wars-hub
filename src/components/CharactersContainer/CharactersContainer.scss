.characters-cont {
    height: 100vh;
    overflow: scroll;
}