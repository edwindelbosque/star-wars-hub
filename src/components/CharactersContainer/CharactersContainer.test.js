import React from 'react';
import { shallow } from 'enzyme';
import CharactersContainer from './CharactersContainer';

describe('CharacterContainer', () => {
  it('component should match the snapshot', () => {
    const wrapper = shallow(
      <CharactersContainer characters={[
        { name: 'Name1', homeworld: 'NameWorld' }
      ]} />
    )
    expect(wrapper).toMatchSnapshot();
  });
})