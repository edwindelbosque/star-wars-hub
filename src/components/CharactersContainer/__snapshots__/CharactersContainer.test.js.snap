// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterContainer component should match the snapshot 1`] = `
<section
  className="characters-cont"
>
  <CharacterCard
    key="0"
    name="Name1"
  />
</section>
`;
